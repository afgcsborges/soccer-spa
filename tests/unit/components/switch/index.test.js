import React from 'react'
import Switch from 'components/switch'
import { act } from 'react-dom/test-utils'
import { shallow } from 'enzyme'

it('Should match snapshot without onChange', async () => {
    const wrapper = shallow(<Switch />)

    act(() => {
        wrapper.find('Styled(Switch)').props().onChange()
    })

    await act(async () => {
        await wrapper.update()
    })

    expect(wrapper).toMatchSnapshot()
})

it('Should toggle switch on click', async () => {
    const mockOnChange = jest.fn()

    const wrapper = shallow(
        <Switch uncheckedLabel={'Unchecked'} checkedLabel={'Checked'} onChange={mockOnChange} checked />
    )

    act(() => {
        wrapper.find('Styled(Switch)').props().onChange()
    })

    await act(async () => {
        await wrapper.update()
    })

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('Text').at(0).props().label).toBe('Unchecked')
    expect(wrapper.find('Text').at(1).props().label).toBe('Checked')
    expect(mockOnChange).toHaveBeenCalled()
})

it('Should toggle switch on click without labels', async () => {
    const mockOnChange = jest.fn()

    const wrapper = shallow(<Switch onChange={mockOnChange} checked />)

    act(() => {
        wrapper.find('Styled(Switch)').props().onChange()
    })

    await act(async () => {
        await wrapper.update()
    })

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('Text')).toMatchObject({})
    expect(mockOnChange).toHaveBeenCalled()
})
