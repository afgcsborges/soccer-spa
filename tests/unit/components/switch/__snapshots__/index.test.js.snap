// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot without onChange 1`] = `
<Space
  align="center"
  direction="horizontal"
  size="middle"
>
  <Styled(Switch)
    onChange={[Function]}
  />
</Space>
`;

exports[`Should toggle switch on click 1`] = `
<Space
  align="center"
  direction="horizontal"
  size="middle"
>
  <Text
    label="Unchecked"
    level={1}
    minWidth="100px"
    strong={false}
    underline={false}
  />
  <Styled(Switch)
    checked={true}
    onChange={[Function]}
  />
  <Text
    label="Checked"
    level={1}
    minWidth="100px"
    strong={true}
    underline={false}
  />
</Space>
`;

exports[`Should toggle switch on click without labels 1`] = `
<Space
  align="center"
  direction="horizontal"
  size="middle"
>
  <Styled(Switch)
    checked={true}
    onChange={[Function]}
  />
</Space>
`;
